### DISCUSSION TITLE
Block in htmx:beforeRequest?

### DISCUSSION THREAD

**Discussion by tobymurray** [bigskysoftware/htmx#1738] [Q&A]
I want to intercept any requests, do some work (await an async function) and then continue with the request. Is that possible? 

Trying with htmx:beforeRequest, I don't believe the request waits for the event to be processed.

**Comment by sheeehy** [comment]
You can use htmx:beforeRequest and then call evt.detail.stopImmediatePropagation() to stop the request while you perform the asynchronous task. Then after your task is complete, you can issue the htmx request using htmx.trigger().

**Reply by tobymurray** [reply]
Awesome - I'll give this a shot and see how it goes!

**Reply by tobymurray** [reply]
I gave this a shot, and it doesn't work directly but inspired what feels like the right path:
```
    event.preventDefault();
    event.stopImmediatePropagation();
```
`stopImmediatePropagation` is not a function on `evt.detail`, and in isolation it did not appear to block the request for me. Also doing `preventDefault()` seems to have done the trick in my context.

**Reply by andryyy** [reply]
I think I used the confirm event a while ago for this. 

https://htmx.org/events/#htmx:confirm
